samples/apps/logging/logging.cpp

```diff
@@ -218,7 +225,120 @@ namespace loggingapp
+        {
+          auto response = std::string("User HTTP signature");
+          response += fmt::format(
+            "\nThe caller is a user with ID: {}", user_sig_ident->user_id);
+          response += fmt::format(
+            "\nThe caller's user data is: {}",
+            user_sig_ident->user_data.dump());
+          response += fmt::format(
+            "\nThe caller's cert is:\n{}", user_sig_ident->user_cert.str());
+
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+          ctx.rpc_ctx->set_response_body(std::move(response));
+          return;
+        }
+        else if (
+          auto member_sig_ident =
+            ctx.template try_get_caller<ccf::MemberSignatureAuthnIdentity>())
+        {
+          auto response = std::string("Member HTTP signature");
+          response += fmt::format(
+            "\nThe caller is a member with ID: {}",
+            member_sig_ident->member_id);
+          response += fmt::format(
+            "\nThe caller's member data is: {}",
+            member_sig_ident->member_data.dump());
+          response += fmt::format(
+            "\nThe caller's cert is:\n{}", member_sig_ident->member_cert.str());
+
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+          ctx.rpc_ctx->set_response_body(std::move(response));
+          return;
+        }
+        else if (
+          auto jwt_ident = ctx.template try_get_caller<ccf::JwtAuthnIdentity>())
+        {
+          auto response = std::string("JWT");
+          response += fmt::format(
+            "\nThe caller is identified by a JWT issued by: {}",
+            jwt_ident->key_issuer);
+          response +=
+            fmt::format("\nThe JWT header is:\n{}", jwt_ident->header.dump(2));
+          response += fmt::format(
+            "\nThe JWT payload is:\n{}", jwt_ident->payload.dump(2));
+
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+          ctx.rpc_ctx->set_response_body(std::move(response));
+          return;
+        }
+        else if (
+          auto no_ident =
+            ctx.template try_get_caller<ccf::EmptyAuthnIdentity>())
+        {
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+          ctx.rpc_ctx->set_response_body("Unauthenticated");
+          return;
+        }
+        else
+        {
+          ctx.rpc_ctx->set_response_status(HTTP_STATUS_INTERNAL_SERVER_ERROR);
+          ctx.rpc_ctx->set_response_body("Unhandled auth type");
+          return;
+        }
+      };
+      make_endpoint("multi_auth", HTTP_GET, multi_auth)
+        .set_auto_schema<void, std::string>()
+        .add_authentication_policy(user_cert_auth_policy)
+        .add_authentication_policy(user_signature_auth_policy)
+        .add_authentication_policy(member_cert_auth_policy)
+        .add_authentication_policy(member_signature_auth_policy)
+        .add_authentication_policy(jwt_auth_policy)
+        .add_authentication_policy(empty_auth_plicy)
```

Comment:
This (and the definition of the endpoint lambda above) are a good starting point - this is what we're trying to support. I think this is a sensible pattern for when there are multiple possible types of identity - try to get each in sequence, process the identity contained with it, potentially execute some common prefix/suffix.

Note that this uses `try_get_caller` (a friendly name for a `dynamic_cast`), since it expects to fail when another auth type was used. Most endpoints will have a single auth type and can safely call `get_caller` (or no auth, in which case they shouldn't call `get_caller` at all).